### PR TITLE
Split the runner environment vars in directories

### PR DESCRIPTION
**Pull request by rtorrero**
When building with the additional checks from `trento-premium` we realized that in order to be able to have multiple `yml` files we would need instead to load a whole directory instead where the additional checks would be placed. But with this approach, the second (alphabetically) loaded file would redefine the `expected` var, causing us to lose all the previously defined expected values provided by default in the community release.

We have come up with a solution to combine both dictionaries into one; see PR#3 on the `trento-premium` repo for further details.

Note: Marking as draft as we would like first some comments about the hardcoded `dir: "/tmp/trento/ansible/vars/` which we don't really like. Using a relative path would make ansible look for the file in `/tmp/trento/ansible/roles/load_facts/vars` which we don't even have.

Co-authored-by: Aleksei Burlakov <aleksei-burlakov@users.noreply.github.com>

### CODE REVIEW

**Comment by arbulu89** [comment]
Hey @rtorrero @aleksei-burlakov ,
This `expected_core` thing looks a pretty hacky patch hehe
Ansible already has options to merge dictionaries as far as a I know. I found this at my 1st search for example: https://stackoverflow.com/questions/25422771/merging-dictionaries-in-ansible

And we can include more ansible roles or operations for that, like the `combine` jinja filter. 

PD: About the hardcoded path, yes, this must be improved. As far as I know ansible uses the same folder of the used tasks file to look for the variables folder, but I don't know if something changes with this `dir` parameter usage

**Comment by rtorrero** [comment]
> Hey @rtorrero @aleksei-burlakov , This `expected_core` thing looks a pretty hacky patch hehe Ansible already has options to merge dictionaries as far as a I know. I found this at my 1st search for example: https://stackoverflow.com/questions/25422771/merging-dictionaries-in-ansible
> 
> And we can include more ansible roles or operations for that, like the `combine` jinja filter.
> 
> PD: About the hardcoded path, yes, this must be improved. As far as I know ansible uses the same folder of the used tasks file to look for the variables folder, but I don't know if something changes with this `dir` parameter usage

The `combine` filter is exactly what we are using, though it's only visible in the `trento-premium` side. What we do here of renaming `expected` to `expected_core` was the only way we found it wouldn't complain while parsing the yaml.

About the hardcoded dir, I thought of just using the config setting; I didn't know that we had this :-)

**Comment by brett060102** [comment]
I will take a look, may be awhile (couple of hours) before I can get to it.

**Comment by rtorrero** [comment]
Just a quick update: @brett060102 provided some comments on slack on how to improve this; I'm testing the proposal and will soon update the PR

**Review by arbulu89 — CHANGES_REQUESTED** [review]
Hey @rtorrero ,
Added a comment on the variables loading.

Besides this, the `expected_core` things is something that doesn't enthusiasm me, it looks a obvious hack hehe
But if we don't have anything better, let's go with it.

With the other fix I would give a green light

File: runner/ansible/roles/load_facts/tasks/main.yml [hunk 1]
```diff
@@ -5,7 +5,8 @@
   register: facts_result
 
 - name: load environment variables
-  include_vars: "{{ env | default('azure') }}_env.yml"
+  include_vars:
+    dir: "{{ lookup('env', 'ANSIBLE_FOLDER') }}/ansible/vars/{{ env | default('azure') }}"
```
[the remark this answers is not in the document]
Comment (rtorrero) [reply runner/ansible/roles/load_facts/tasks/main.yml:9]: Ok, this simplifies things a lot!

**Review by dottorblaster — APPROVED** [review]
Once @arbulu89's comment is addressed, this feels super good to me! Go go go

File: runner/ansible/roles/load_facts/tasks/main.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (rtorrero) [reply runner/ansible/roles/load_facts/tasks/main.yml]: thanks!

**Pull request merged**